doc/design.txt

```diff
@@ -210,15 +210,15 @@ An image trailer has the following structure:
      0                   1                   2                   3
```

Comment:
> I'm enumerating all of the states and asking what the boot outcome would be in each one, which is a perfectly correct thing to do when considering "possible" states.
> 
> A decision tree representation is equivalent to a complete enumeration

Further, the document above is not described as a decision tree, but rather an enumeration of states. Perhaps if that's what the code is really doing, then reworking the "BOOT STATES" section so it's described a decision tree is what is needed? Then we can compare that to how the concrete "vector states" (with their "Any" values) map to that tree, to convince ourselves things work the way we mean them to.